### PR TITLE
Release candidate 2.0.5

### PR DESCRIPTION
**Pull request by ferras**
[Add reliability improvements](https://github.com/Netflix/metaflow-service/pull/150)
[DB schema update to distinguish between metadata from different attempts](https://github.com/Netflix/metaflow-service/pull/30)
[refactoring in preparation for ui backend](https://github.com/Netflix/metaflow-service/pull/144)
[Add tests](https://github.com/Netflix/metaflow-service/pull/145)

### CODE REVIEW

File: services/metadata_service/api/utils.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    return wrapper
+
+
+def web_response(status: int, body):
+    return web.Response(status=status,
+                        body=json.dumps(body),
+                        headers=MultiDict(
+                            {"Content-Type": "application/json",
+                             "Access-Control-Allow-Origin": "*",
+                             METADATA_SERVICE_HEADER: METADATA_SERVICE_VERSION}))
```
Comment (saikonen) [inline services/metadata_service/api/utils.py:38]: as this is now only used for the metadata service responses, we can probably remove the access-control headers as unnecessary.

File: services/metadata_service/api/utils.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        except Exception as err:
+            # pass along an id for the error
+            err_id = getattr(err, 'id', None)
+            # either use provided traceback from subprocess, or generate trace from current process
+            err_trace = getattr(err, 'traceback_str', None) or get_traceback_str()
+            if not err_id:
+                # Log error only in case it is not a known case.
+                err_id = 'generic-error'
+                logging.error(err_trace)
+            return http_500(str(err), err_id, err_trace)
```
Comment (saikonen) [inline services/metadata_service/api/utils.py]: the passing of an error ID is specific to the UI service which uses custom error classes for some cases. I think this functionality could be pruned down a bit for the metadata service which does not implement any custom-id errors

File: services/utils/tests/unit_tests/utils_test.py [hunk 3]
```diff
@@ -3,7 +3,8 @@
 import contextlib
 import json
 from aiohttp.test_utils import make_mocked_request
-from services.utils import format_qs, format_baseurl, DBConfiguration, handle_exceptions
+from services.utils import format_qs, format_baseurl, DBConfiguration
+from services.metadata_service.api.utils import handle_exceptions
```
Comment (saikonen) [inline services/utils/tests/unit_tests/utils_test.py:7]: speaking of circular dependencies, now that the `handle_exceptions` was moved to the metadata services api/utils, consider moving the corresponding tests to `services/metadata_service/tests/unit_tests/` instead.

File: MANIFEST.in [hunk 4]
```diff
@@ -0,0 +1,5 @@
+include services/utils/requirements.txt
+include services/data/requirements.txt
```
Comment (saikonen) [inline MANIFEST.in:2]: these two files got deleted in the dependency unpinning commit.

File: requirements.txt [hunk 5]
```diff
@@ -0,0 +1,4 @@
+-r services/utils/requirements.txt
+-r services/data/requirements.txt
```
Comment (saikonen) [inline requirements.txt:2]: these two files got deleted in the dependency unpinning commit. What was the reason for completely removing module specific dependencies for the data and utils?

File: services/metadata_service/api/utils.py (hunk 2 above)
Comment (ferras) [reply services/metadata_service/api/utils.py]: rolled back to original version of `handle_exceptions`

File: MANIFEST.in (hunk 4 above)
Comment (ferras) [reply MANIFEST.in:2]: removed

File: requirements.txt (hunk 5 above)
Comment (ferras) [reply requirements.txt:2]: removed

File: services/metadata_service/tests/unit_tests/api_util_test.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    async def do_not_raise():
+        return True
+
+    @format_response
+    @handle_exceptions
+    async def raise_without_id():
+        raise Exception("test")
+
+    # wrapper should not touch successful calls.
+    assert (await do_not_raise())
```
Comment (saikonen) [inline services/metadata_service/tests/unit_tests/api_util_test.py:16]: I'm guessing the addition of `@format_response` was to get this test to pass? It has the unwanted effect that we are also testing the format_response implementation in a single unit test. Not a blocker for merging of course, and can be addressed in #146 later on.

Comment (ferras) [reply services/metadata_service/tests/unit_tests/api_util_test.py:16]: considering `@handle_exceptions` is only ever used when accompanied by  `@format_response` I don't think this is a problem.

**Pull request merged**